File: LC_JS-SOLUTIONS/205. Isomorphic Strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function (s, t) {
  let sDic = {};
  let tDic = {};

  for (let i = 0; i < s.length; i++) {
    let charS = s[i];
    let charT = t[i];
    if (charS in sDic) {
      if (sDic[charS] !== charT) return false;
    } else {
      sDic[charS] = charT;
    }

    if (charT in tDic) {
      if (tDic[charT] !== charS) return false;
    } else {
      tDic[charT] = charS;
    }
  }

  return true;
};

File: LC-Python-Solution/205. Isomorphic Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:

        sDic = {}
        tDic = {}

        for i in range(len(s)):
            charS = s[i]
            charT = t[i]

            if charS in sDic:
                if sDic[charS] != charT:
                    return False
            else:
                sDic[charS] = charT

            if charT in tDic:
                if tDic[charT] != charS:
                    return False
            else:
                tDic[charT] = charS
        
        return True

        

